core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionValidityService.js

```diff
@@ -0,0 +1,24 @@
+// Service for keeping track of solution validity.
+oppia.factory('SolutionValidityService', [
```

Comment:
Frontend tests (spec files). I think there are existing examples of how to test services, they should actually be fairly straightforward to write once you've set up the boilerplate.